File: internal/content/testdata/bench/go/nc-walls-and-gates.go
func walls_and_gates(rooms [][]int) [][]int {
	if len(rooms) == 0 || len(rooms[0]) == 0 {
		return rooms
	}
	const INF = 2147483647
	m, n := len(rooms), len(rooms[0])
	type point struct{ r, c int }
	queue := []point{}
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if rooms[i][j] == 0 {
				queue = append(queue, point{i, j})
			}
		}
	}
	dirs := []point{{1, 0}, {-1, 0}, {0, 1}, {0, -1}}
	for len(queue) > 0 {
		cur := queue[0]
		queue = queue[1:]
		for _, d := range dirs {
			nr, nc := cur.r+d.r, cur.c+d.c
			if nr >= 0 && nr < m && nc >= 0 && nc < n && rooms[nr][nc] == INF {
				rooms[nr][nc] = rooms[cur.r][cur.c] + 1
				queue = append(queue, point{nr, nc})
			}
		}
	}
	return rooms
}

File: internal/content/testdata/bench/java/nc-walls-and-gates.java
import java.util.*;

class Solution {
    public long[][] walls_and_gates(long[][] rooms) {
        if (rooms == null || rooms.length == 0 || rooms[0].length == 0) return rooms;
        long INF = 2147483647L;
        int m = rooms.length, n = rooms[0].length;
        Queue<int[]> queue = new LinkedList<>();
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (rooms[i][j] == 0) {
                    queue.add(new int[]{i, j});
                }
            }
        }
        int[][] dirs = {{1,0},{-1,0},{0,1},{0,-1}};
        while (!queue.isEmpty()) {
            int[] cell = queue.poll();
            int i = cell[0], j = cell[1];
            for (int[] d : dirs) {
                int ni = i + d[0], nj = j + d[1];
                if (ni >= 0 && ni < m && nj >= 0 && nj < n && rooms[ni][nj] == INF) {
                    rooms[ni][nj] = rooms[i][j] + 1;
                    queue.add(new int[]{ni, nj});
                }
            }
        }
        return rooms;
    }
}
